plugins/chat/spec/system/network_reconciliation_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+RSpec.describe "Network reconciliation", type: :system, js: true do
+  fab!(:current_user) { Fabricate(:user) }
+  fab!(:other_user) { Fabricate(:user) }
+  fab!(:channel_1) { Fabricate(:category_channel) }
+
+  let(:chat_page) { PageObjects::Pages::Chat.new }
+  let(:chat_channel_page) { PageObjects::Pages::ChatChannel.new }
+
+  before do
+    chat_system_bootstrap
+    channel_1.add(current_user)
+    channel_1.add(other_user)
+  end
+
+  context "when user recovers network" do
+    it "recovers state" do
+      using_session(:disconnected_current_user) do
+        sign_in(current_user)
+        visit("/")
+        page.driver.browser.network_conditions = { offline: true, latency: 0, throughput: 0 }
```

Comment:
Any reason to not also have `session.quit` here?

Reply:
Thanks!

Yes because we want the browser tab to stay opened with network of, otherwise we will have to reopen a tab and load a page which will defeat the purpose of the test.

Reply:
If you want to run the test locally, one cool trick you can is this with a different color inside each session:

```
page.evaluate_script("document.body.style.backgroundColor = 'red';")
```

Reply:
Oh that is cool!